### ISSUE TITLE
False positive for react/prop-types in namespaced components

### ISSUE THREAD

**Issue by akozhemiakin** [yannickcr/eslint-plugin-react#621]
```jsx
const SideMenu = ({ children }) => (
  <div className="c-side-menu">
    {children}
  </div>);

SideMenu.propTypes = {
  children: React.PropTypes.node,
};

SideMenu.Item = ({ onClick, children }) => (
  <div className="item" onClick={onClick}>{children}</div>
);

SideMenu.Item.propTypes = {
  onClick: React.PropTypes.func,
  children: React.PropTypes.node,
};
```
Here eslint complains about missing prop types validation for SideMenu.Item

**Comment by ljharb** [comment]
Why is it a false positive? `SideMenu.Item` is a component, and it needs `propTypes`.

**Comment by akozhemiakin** [comment]
Yes, and as you can see in the provided code chunk propTypes are defined on it. But it seems that eslint does not see it.

**Comment by ljharb** [comment]
aha, thanks for clarifying :-)
